File: frontend/frontend_idea_app/lib/models/Comment.dart
class Comment {
  String author, content, timestamp;
  int vote;

  Comment(this.content, this.author, this.timestamp, this.vote);

  Comment.fromJson(Map<String, dynamic> json)
      : content = json['content'],
        author = json['author'],
        timestamp = json['timestamp'],
        vote = json['vote'];

  Map<String, dynamic> toJson() => {
        "content": "'" + content + "'",
        "author": "'" + author + "'",
        "timestamp": "'" + timestamp + "'",
        "vote": vote
      };
}

File: backend/ideappservice/src/main/java/com/ideapp/ideappservice/model/Comment.java
package com.ideapp.ideappservice.model;

public class Comment {
    String content;
    String author;
    String timestamp;
    int vote;

    public Comment(){}

    public Comment(String content, String author, String timestamp,int vote) {
        this.content = content;
        this.vote = vote;
        this.author = author;
        this.timestamp = timestamp;
    }

    public String getContent() {
        return content;
    }
    

    public void setComment(String content) {
        this.content = content;
    }

    public int getVote() {
        return vote;
    }
    

    public void setVote(int vote) {
        this.vote = vote;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}
